### PR TITLE
fix issues and add responsivesness

### PR DESCRIPTION
**Pull request by zara19013**

### CODE REVIEW

File: src/components/homepage/filter.js [hunk 1]
```diff
@@ -49,13 +46,33 @@ function Filter({ countriesData, onFilterChange }) {
...
               <ul className="selectList">
-                <li data-value="All" onClick={handleChange}> All</li>
-                <li data-value="Africa" onClick={handleChange}> Africa</li>
-                <li data-value="Americas" onClick={handleChange}> America</li>
-                <li data-value="Antarctic" onClick={handleChange}> Antarctic</li>
-                <li data-value="Asia" onClick={handleChange}> Asia</li>
-                <li data-value="Europe" onClick={handleChange}> Europe</li>
-                <li data-value="Oceania" onClick={handleChange}> Oceania</li>
+                <li data-value="All" onClick={handleChange}>
+                  All
```
Comment (asad-shayansolution) [inline src/components/homepage/filter.js]: Make an array of All filter values and use Array.map method to render this list

**Review by asad-shayansolution — COMMENTED** [review]
When I click onto some flag the application breaks.

File: src/components/homepage/filter.js [hunk 2]
```diff
@@ -1,45 +1,42 @@
...
+    setIsOpen(!isOpen);
+  }
+
+  function handleChange(e) {
+    const region = e.target.getAttribute("data-value");
+
+    if (region === "All") {
+      onFilterChange(countriesData);
+    } else {
+      fetch(`${apiBaseUrl}region/${region}`)
```
Comment (asad-shayansolution) [inline src/components/homepage/filter.js]: Make an api folder into src/ directory and add an /api/ folder.
All the api calls would be written into that api.js file.

**Review by asad-shayansolution — CHANGES_REQUESTED** [review]

File: src/components/homepage/filter.js (hunk 1 above)
Comment (zara19013) [reply src/components/homepage/filter.js]: updated

**Review by imran-munawar — APPROVED** [review]

File: src/components/homepage/filter.js (hunk 2 above)
Comment (zara19013) [reply src/components/homepage/filter.js]: updated

**Review by asad-shayansolution — APPROVED** [review]

**Pull request merged**